xrpl4j-model/src/main/java/org/xrpl/xrpl4j/model/client/accounts/NfTokenObject.java

```diff
@@ -0,0 +1,33 @@
+package org.xrpl.xrpl4j.model.client.accounts;
+
+//import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+//import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.google.common.primitives.UnsignedLong;
+import org.immutables.value.Value;
+import org.xrpl.xrpl4j.model.flags.Flags;
+import org.xrpl.xrpl4j.model.transactions.NfTokenMint;
+
+import java.util.Optional;
+
+/**
+ * Structure of an NFToken stored on the ledger.
+ */
+@Value.Immutable
+//@JsonSerialize(as = ImmutableNfToken.class)
+//@JsonDeserialize(as = ImmutableNfToken.class)
+public interface NfTokenObject {
+
+  /**
+   * The unique TokenID of the token.
+   *
+   * @return The unique TokenID of the token.
+   */
+  String tokenId();
```

Comment:
We should create a `TokenId` type here -- see `Ids.java`

Reply:
xls-20d states the following about `TokenID`:
> This composite field uniquely identifiers a token; it contains a set of 16 bits that identify flags or settings specific to the NFT, 16 bits that encode the transfer fee associated with this token, if
any, the 160-bit account identifier of the issuer, a 32-bit issuer-specified taxon, and an
(automatically generated) monotonically increasing 32-bit sequence number.

Do we want to try to parse this ID in our `TokenId` wrapper class or are those bits of info available by other means?

Reply:
I'm not opposed to parsing meaning if we think it's useful. We _could_ do this in the Id wrapper, but given there's a lot of meaning, it might make sense to create a dedicated class.